enterprise/server/remote_execution/runner/runner.go

```diff
@@ -165,6 +165,12 @@ func (r *CommandRunner) PrepareForTask(task *repb.ExecutionTask) error {
 	if err := r.Workspace.CreateOutputDirs(); err != nil {
 		return status.UnavailableErrorf("Error creating output directory: %s", err.Error())
 	}
+
+	// Pull the container image before Run() is called, so that we don't
+	// use up the whole exec ctx timeout with a slow container pull.
+	if err := r.Container.PullImageIfNecessary(ctx); err != nil {
+		return status.UnavailableErrorf("Error pulling container: %s", err)
+	}
```

Comment:
Maybe add a comment to the `Run` func that the `PullImageIfNecessary` should be a NOP since we pull in `PrepareTask`? (I understand not wanting to remove the pull from that func, but might be confusing to future readers why we pull twice)